lib/vrl/compiler/src/expression.rs

```diff
@@ -103,16 +122,24 @@ clone_trait_object!(Expression);
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum Expr {
+    #[cfg(feature = "expr-literal")]
     Literal(Literal),
     Container(Container),
+    #[cfg(feature = "expr-if_statement")]
     IfStatement(IfStatement),
+    #[cfg(feature = "expr-op")]
     Op(Op),
+    #[cfg(feature = "expr-assignment")]
     Assignment(Assignment),
+    #[cfg(feature = "expr-query")]
     Query(Query),
+    #[cfg(feature = "expr-function_call")]
     FunctionCall(FunctionCall),
     Variable(Variable),
```

Comment:
I didn't add an `expr-variable` feature, as I plan to factor this type out in a follow-up PR, as it's a subset of the `query` expression.